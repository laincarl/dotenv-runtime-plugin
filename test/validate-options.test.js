import DotEnvRuntimePlugin from "../lib";

it("validation", () => {
  /* eslint-disable no-new */
  expect(() => {
    new DotEnvRuntimePlugin({ entry: "./.env" });
  }).not.toThrow();

  expect(() => {
    new DotEnvRuntimePlugin({ entry: ".env" });
  }).not.toThrow();

  expect(() => {
    new DotEnvRuntimePlugin({ test: "foo" });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new DotEnvRuntimePlugin({ extra: "foo" });
  }).toThrowErrorMatchingSnapshot();
});
