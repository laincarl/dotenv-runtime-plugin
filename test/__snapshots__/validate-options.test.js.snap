// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Invalid options object. DotEnvRuntime Plugin has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'test'. These properties are valid:
   object { entry? }"
`;

exports[`validation 2`] = `
"Invalid options object. DotEnvRuntime Plugin has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'extra'. These properties are valid:
   object { entry? }"
`;
